src/protocol.c

```diff
@@ -568,7 +568,9 @@ int trilogy_build_auth_packet(trilogy_builder_t *builder, const char *user, cons
 
     if (pass_len > 0) {
         // Fallback to te default unless we have SHA2 requested
```

Comment:
nit: I think this comment should likely be moved down to between line 573/4 and another comment should be added here to add context to the new behavior.